src/lib.rs

```diff
@@ -228,9 +230,28 @@ impl Datadog<dogstatsd::Client> {
     }
 
     /// Time a block of code (reports in ms)
-    pub fn time<S: AsRef<str>>(metric: impl AsRef<str>, tags: impl TagsProvider<S>, block: impl FnOnce()) {
+    pub fn time<S, F, O>(metric: impl AsRef<str>, tags: impl TagsProvider<S>, block: F) -> O
+    where
+        S: AsRef<str>,
+        F: FnOnce() -> O,
+    {
+        if let Some(instance) = INSTANCE.get() {
+            instance.do_time(metric.as_ref(), tags, block)
+        } else {
+            block()
+        }
+    }
+
+    pub async fn async_time<S, F, T, O>(metric: &str, tags: impl TagsProvider<S> + Send + Sync, block: F) -> O
+    where
+        S: AsRef<str> + Sync,
+        F: FnOnce() -> T + Send,
+        T: Future<Output = O> + Send,
+    {
         if let Some(instance) = INSTANCE.get() {
-            instance.do_time(metric.as_ref(), tags, block);
+            instance.do_async_time(metric.as_ref(), tags, block).await
+        } else {
+            block().await
```

Comment:
Was this missing `else` a bug before...?

Reply:
Looking at it, I think so - you could make an argument that if the library isn't initialized then blocks passed to it shouldn't do anything, but I think that behavior would be surprising.